queries/wing/highlights.scm

```diff
@@ -100,6 +155,26 @@
   "pub"
   "protected"
   "internal"
+  "extern"
+  (static)
 ] @keyword.modifier
 
 "return" @keyword.return
+
+(import_statement
+  module_name: (identifier) @module)
+
+(import_statement
+  alias: (identifier) @module)
+
+(call
+  (reference
+    (nested_identifier
+      object: (reference) @_ref
+      property: (member_identifier) @_ident))
+  (argument_list
+    (positional_argument
+      (string) @string.regexp))
+  (#eq? @_ref "regex")
+  (#eq? @_ident "compile")
+  (#offset! @string.regexp 0 1 0 -1))
```

Comment:
While this is very cool I think highlighting implies that the string is special or is handled in a special way (e.g. different escape sequences) when it's really just a normal string.

Reply:
No, you just get "smarter" highlighting for free; it's still a string in the end. Unless the language has no support at all for regular expressions?

Reply:
It has no support at a language level, no. The standard library has a type that allows you to write `regex.compile("^[0-1]*$")` but the string there is a simple string, it can also be interpolated or you can also pass a variable.

Reply:
Still, users would like to have the regex highlighted (consistently) to make it easier to parse -- which is what this parser injection ensures.

(Remember, the goal here is to use tree-sitter for better and more consistent syntax highlighting, not linting or otherwise enforcing an abstract grammar.)

Reply:
> it can also be interpolated or you can pass a variable.

But if there's a serious risk of mis-parsing, that's different of course.

Reply:
Yeah the risk is misparsing which will be misleading to users, specifically with regards to the interpolation braces `{}` and escape sequences.
Wing also has a non-interpolated string literal, but escape sequences will still be a problem there.

Reply:
Then the function name is misleading, too ;) If you can't assume that the argument of  `regex.compile` is a regular expression, something has gone wrong in your standard library design...

Reply:
The function will throw if the argument is not a valid regex string. There's no actual intrinsic regex type nor syntax, so there's no way to convey that more strictly in the type signature. Wing is still an early language so that's all subject to change.

Reply:
Again, the purpose here is _not_ linting, that is out-of-scope. We want to correctly highlight _correct_ code and do not care about incorrect code. So just because the function _can_ take non-regex and give an error instead of crashing, that is _not_ sufficient reason to not give the (expected) argument a more useful highlight.

The situation is different if users are _expected_ to put in a non-regexp string in a _correct_ situation (in which case the function name is weird, but you do do you).

Reply:
I fully understand that, I'm trying to convey that wing's strings with interpolation and escapes will both look and become valid regex, but the highlights will be misleading:

```wing
let someText = "hi";
let validRegex = regex.compile("{someText[0]}+");
```

In this example, the final regex will actually be `/h+/` due to interpolation, but the highlighting will make it seem like the regex is `/{someText[0]}+/ because it's seeing it all as a regex.

Reply:
An example says more than a thousand words :) Yes, then this injection is unsafe and should not be added.

Reply:
I'm not desperate to add this, but I think the injection is still safe: interpolations are exposed under child of string (`(template_substitution)`) and thus the injection is not applied to them. The only weird case I can think of would be something like `regex.compile("{var}-9]")` where `var` is something like `"[0"` (causing unmatched brackets from the injection's point of view), but I imagine cases like that would be insanely uncommon. If this is still to dangerous, I don't mind removing it

Reply:
> interpolations are exposed under child of string and thus the injection is not applied to them

Ah ok nice, I didn't know injections worked like that. In that case I'm cool with this and it should be fine with escape sequences as well. 👍